generator/src/main/java/com/scottlogic/deg/generator/outputs/TestCaseDataSet.java

```diff
@@ -1,23 +1,23 @@
 package com.scottlogic.deg.generator.outputs;
 
+import com.scottlogic.deg.generator.constraints.Constraint;
 import com.scottlogic.deg.generator.inputs.RuleInformation;
 
+import java.util.function.Supplier;
 import java.util.stream.Stream;
 
 public class TestCaseDataSet {
-    public final RuleInformation violation;
-    private final Stream<GeneratedObject> rows;
+    public final RuleInformation violatedRule;
+    public final Constraint violatedConstraint;
+    private final Supplier<Stream<GeneratedObject>> rows;
```

Comment:
This change was to support laziness with row generation, something is happening with the creation of `TestCaseDataSet` when it holds a `Stream` as a field. Might be that the stack is getting confused through its laziness, difficult to tell.
This could be reverted, but for clarity when debugging in other scenarios it might be an idea to keep it.